README.md

```diff
@@ -381,6 +381,19 @@ int main(int argc, char* argv[])
 }
 ```
 
+## Contributor Guidelines
+* This repo supports [pre-commit](https://pre-commit.com/) e.g. for automatic code formatting. TLDR:
+  This will prevent you from committing falsely formatted code:
+  ```
+  pipx install pre-commit
```

Comment:
I think we should tell them which clang format to install as well. 
```
```suggestion
  pipx install pre-commit 
  sudo apt install clang-format-14
```

Reply:
The clang-format version is specified in the pre-commit job. That's the main reason I switched to pre-commit: clang-format is installed in a python virtualenv with the respective version, so developers (and CI maintainers) don't have to manually install a specific version.

Reply:
I wasn't aware of that, nice solution :)